isce2_topsapp/iono_proc.py

```diff
@@ -0,0 +1,383 @@
+        image.setDataType('FLOAT')
+        image.setFilename(ionFiltOut)
+        image.extraFilename = ionFiltOut + '.vrt'
+        image.setWidth(img.width)
+        image.setLength(img.length)
+        # image.setAccessMode('read')
+        # image.createImage()
+        image.renderHdr()
+        # image.finalizeImage()
+
+
+####################### UTILITIES FOR MASKING ###############################
+
+def mask_geo2radar(maskFilename: Union[str, Path],
+                   latFilename: Union[str, Path],
+                   lonFilename: Union[str, Path],
+                   outputFilename: Union[str, Path],
+                   saveFlag: bool = False) -> NDArray:
+    '''
+    This routine translates mask raster from geographical to radar(image) coordinate space
+
+    maskFilename : str
+            path to georeferenced mask raster (must have vrt extension)
+    latFilename : str
+            path to radar lat.rdr file (bursts: geom_reference/IW{1,2,3}/lat_burst.rdr
+                                        merged: merged/lat.rdr.full.vrt)
+    lonFilename : str
+            path to radar lon.rdr file (bursts: geom_reference/IW{1,2,3}/lon_burst.rdr
+                                        merged: merged/lon.rdr.full.vrt)
+    outputFilename : str
+            path to save mask output file
+    saveFlag : bool
+            flag to locally save mask in radar coordinates
+    '''
+
+    # Open mask file
+    mask_ds = gdal.Open(maskFilename + '.vrt')
+    # Open lon and lat file
+    lon_ds = gdal.Open(lonFilename + '.vrt')
+    lat_ds = gdal.Open(latFilename + '.vrt')
+
+    # Get lon and lat arrays
+    lons = lon_ds.ReadAsArray()
+    lats = lat_ds.ReadAsArray()
+
+    # Translate mask geo corrdinate to radar
+    lineIndex = np.int32((lats - mask_ds.GetGeoTransform()[3]) / mask_ds.GetGeoTransform()[5] + 0.5)
+    sampleIndex = np.int32((lons - mask_ds.GetGeoTransform()[0]) / mask_ds.GetGeoTransform()[1] + 0.5)
+    inboundIndex = np.logical_and(
+                np.logical_and(lineIndex >= 0, lineIndex <= mask_ds.RasterYSize-1),
+                np.logical_and(sampleIndex >= 0, sampleIndex <= mask_ds.RasterXSize-1)
+                )
+    # Convert
+    mask_radarcoord = np.empty(lats.shape)
+    mask_radarcoord[inboundIndex] = mask_ds.ReadAsArray()[lineIndex[inboundIndex],
+                                                            sampleIndex[inboundIndex]]
+    # close gdal instances
+    mask_ds = None
+    lon_ds = None
+    lat_ds = None
+
+    # Save
+    if saveFlag:
+        mask_radarcoord.astype(np.int8).tofile(outputFilename)
+        image = isceobj.createImage()
+        image.setDataType('BYTE')
+        image.setFilename(outputFilename)
+        image.extraFilename = outputFilename + '.vrt'
+        image.setWidth(mask_radarcoord.shape[1])
+        image.setLength(mask_radarcoord.shape[0])
+        image.renderHdr()
+
+    return mask_radarcoord
+
+
+def mask_interferogram(ifgFilename: Union[str, Path],
```

Comment:
you could potentially also name this `mask_raster_in_rdr` or whatever you like since again this could be repurposed.